File: hseolymp/2016-17/qualifying_round/demo/power_socket.cpp
#include <iostream>

int main()
{
	int n, m, a, b;
	std::cin >> n >> m >> a >> b;
	std::cout << ((4*a < b) ? (m - n) * a : (m - n) / 4 * b + std::min((m - n) % 4 * a, b)) << std::endl;
	return 0;
}

File: hseolymp/2016-17/qualifying_round/demo/power_socket.py
n = int(input())
m = int(input())
a = int(input())
b = int(input())

lack = m - n

if lack <= 0:
    print(0)
else:
    if a <= b/4:
        print(lack*a)
    else:
        multi = lack // 4
        lack %= 4
        print(multi*b + min(a*lack, b))
